### ISSUE TITLE
realsense-viewer - Could not initialize offscreen context!

### ISSUE THREAD

**Issue by Borillion** [datasith/Ai_Demos_RPi#1]
Hi I am curious if you have seen this issue before?
When I try to run realsense-viewer, it outputs Could not initialize offscreen context!

Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

Linux raspberrypi 4.19.118-v7l+ #1311 SMP Mon Apr 27 14:26:42 BST 2020 armv7l GNU/Linux

pi@raspberrypi:~ $ realsense-viewer
Could not initialize offscreen context!
pi@raspberrypi:~ $ ldd $(which realsense-viewer)
	linux-vdso.so.1 (0xbef2c000)
	/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0xb6fb2000)
	libatomic.so.1 => /usr/lib/arm-linux-gnueabihf/libatomic.so.1 (0xb6f81000)
	libGL.so.1 => /usr/lib/arm-linux-gnueabihf/libGL.so.1 (0xb6ef4000)
	libGLU.so.1 => /usr/lib/arm-linux-gnueabihf/libGLU.so.1 (0xb6e8c000)
	librealsense2-gl.so.2.34 => /usr/local/lib/librealsense2-gl.so.2.34 (0xb6c1a000)
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb6c03000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6bf0000)
	libX11.so.6 => /usr/lib/arm-linux-gnueabihf/libX11.so.6 (0xb6ace000)
	librealsense2.so.2.34 => /usr/local/lib/librealsense2.so.2.34 (0xb5a96000)
	libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb594f000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb58cd000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb58a0000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb5876000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb5728000)
	libGLdispatch.so.0 => /usr/lib/arm-linux-gnueabihf/libGLdispatch.so.0 (0xb56a9000)
	libGLX.so.0 => /usr/lib/arm-linux-gnueabihf/libGLX.so.0 (0xb568a000)
	/lib/ld-linux-armhf.so.3 (0xb6fc7000)
	libxcb.so.1 => /usr/lib/arm-linux-gnueabihf/libxcb.so.1 (0xb565b000)
	libusb-1.0.so.0 => /lib/arm-linux-gnueabihf/libusb-1.0.so.0 (0xb5636000)
	libXau.so.6 => /usr/lib/arm-linux-gnueabihf/libXau.so.6 (0xb5623000)
	libXdmcp.so.6 => /usr/lib/arm-linux-gnueabihf/libXdmcp.so.6 (0xb560e000)
	libudev.so.1 => /lib/arm-linux-gnueabihf/libudev.so.1 (0xb55de000)
	libbsd.so.0 => /usr/lib/arm-linux-gnueabihf/libbsd.so.0 (0xb55b6000)

pi@raspberrypi:~ $ echo $LD_LIBRARY_PATH
/usr/local/lib:
pi@raspberrypi:~ $ echo $PYTHONPATH
:/usr/local/lib

**Comment by datasith** [comment]
heya, i haven't seen it before. are your running the viewer connected to a monitor or over ssh?

**Comment by Borillion** [comment]
Is directly over ssh/vpn, I am using the latest Raspberry Pi OS. Thought it
was weird. I can run glxgears and glxinfo fine too.

**Comment by Borillion** [comment]
Do you mind running` ldd $(which realsense-viewer) `for me to compare ?

**Comment by datasith** [comment]
I tried `export` the correct `DISPLAY` environment variable to no avail. So I settled for doing `VNC` when using their viewer, and that works w/o problem.

here it is:

```
	linux-vdso.so.1 (0xbedf2000)
	/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0xb6eba000)
	libatomic.so.1 => /usr/lib/arm-linux-gnueabihf/libatomic.so.1 (0xb6e8b000)
	libGL.so.1 => /usr/lib/arm-linux-gnueabihf/libGL.so.1 (0xb6dfe000)
	libGLU.so.1 => /usr/lib/arm-linux-gnueabihf/libGLU.so.1 (0xb6d96000)
	librealsense2-gl.so.2.35 => /usr/local/lib/librealsense2-gl.so.2.35 (0xb6b22000)
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb6b0b000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6af8000)
	libX11.so.6 => /usr/lib/arm-linux-gnueabihf/libX11.so.6 (0xb69d6000)
	librealsense2.so.2.35 => /usr/local/lib/librealsense2.so.2.35 (0xb5997000)
	libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb5850000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb57ce000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb57a1000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb5777000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb5629000)
	libGLdispatch.so.0 => /usr/lib/arm-linux-gnueabihf/libGLdispatch.so.0 (0xb55aa000)
	libGLX.so.0 => /usr/lib/arm-linux-gnueabihf/libGLX.so.0 (0xb558b000)
	/lib/ld-linux-armhf.so.3 (0xb6ecf000)
	libxcb.so.1 => /usr/lib/arm-linux-gnueabihf/libxcb.so.1 (0xb555c000)
	libusb-1.0.so.0 => /lib/arm-linux-gnueabihf/libusb-1.0.so.0 (0xb5537000)
	libXau.so.6 => /usr/lib/arm-linux-gnueabihf/libXau.so.6 (0xb5524000)
	libXdmcp.so.6 => /usr/lib/arm-linux-gnueabihf/libXdmcp.so.6 (0xb550f000)
	libudev.so.1 => /lib/arm-linux-gnueabihf/libudev.so.1 (0xb54df000)
	libbsd.so.0 => /usr/lib/arm-linux-gnueabihf/libbsd.so.0 (0xb54b7000)
```

HTH!

**Comment by Borillion** [comment]
So the difference I see between yours and mine off the top is
You:
	librealsense2.so.2.35 => /usr/local/lib/librealsense2.so.2.35 (0xb5997000)
Me: 
        librealsense2.so.2.34 => /usr/local/lib/librealsense2.so.2.34 (0xb5a96000)

I will try and see if thats it.

**Comment by Borillion** [comment]
No it wasn't that. :) appears there is a difference between running vncserver from the terminal and using raspi-config to install it. 

If you see  the below output your doing it wrong!
pi@raspberrypi:~ $ glxgears -info
GL_RENDERER   = llvmpipe (LLVM 9.0.1, 128 bits)
GL_VERSION    = 3.1 Mesa 19.3.2
GL_VENDOR     = VMware, Inc.

Output thats working for me
GL_RENDERER   = V3D 4.2
GL_VERSION    = 2.1 Mesa 19.3.2
GL_VENDOR     = Broadcom

**Comment by datasith** [comment]
oh dang, it would've taken me a while to sort that one out. i'll add it to the wiki shortly. i'll leave the issue open as a reminder! thanks for the help.

**Comment by Borillion** [comment]
Yea no problem!

**Comment by bestaps** [comment]
> No it wasn't that. :) appears there is a difference between running vncserver from the terminal and using raspi-config to install it.
> 
> If you see the below output your doing it wrong!
> pi@raspberrypi:~ $ glxgears -info
> GL_RENDERER = llvmpipe (LLVM 9.0.1, 128 bits)
> GL_VERSION = 3.1 Mesa 19.3.2
> GL_VENDOR = VMware, Inc.
> 
> Output thats working for me
> GL_RENDERER = V3D 4.2
> GL_VERSION = 2.1 Mesa 19.3.2
> GL_VENDOR = Broadcom

hey @Borillion, I am also seeing GL_VENDOR = VMware, Inc. any documentation on how you changed it to GL_VENDOR = Broadcom. thanks 

Edit:- resolved it by following @Borillion on raspberrypi/Raspberry-Pi-OS-64bit#21 and raspberrypi/Raspberry-Pi-OS-64bit#3 thanks

**Comment by bestaps** [comment]
@themakerbro 
adding to the error could not initialize offscreen context I am also facing GLFW Driver error. any ideas on resolving it ? 

Linux raspberrypi 5.4.61-v7l+ #1339 SMP Tue Sep 1 18:51:27 BST 2020 armv7l GNU/Linux

pi@raspberrypi:~ $ realsense-viewer
GLFW Driver Error: GLX: GLX version 1.3 is required
Could not initialize offscreen context!

Edit:- Resolved it somehow IntelRealSense/librealsense#7343

**Comment by datasith** [comment]
After several months, finally added the info to the wiki!

https://github.com/acrobotic/Ai_Demos_RPi/wiki/Raspberry-Pi-4-and-Intel-RealSense-D435
